report/report.go

```diff
@@ -6,21 +6,12 @@ import (
 	"time"
 )
 
-// Report is the internal structure produced and emitted by the probe, and
-// operated-on (e.g. merged) by intermediaries and the app. The probe may fill
-// in as many topologies as it's capable of producing, including none.
-//
-// Process, [Transport,] and Network topologies are distinct because the data
-// sources are distinct. That is, the Process topology can only be populated
-// by extant connections between processes, but the Network topology may be
-// populated from e.g. system-level data sources.
-//
-// Since the data sources are fundamentally different for each topology, it
-// might make sense to make them more distinct in the user interface.
+// Report is the core data type. It's produced by probes, and consumed and
+// stored by apps. It's composed of multiple topologies, each representing
+// a different (related, but not equivalent) view of the network.
 type Report struct {
-	Process Topology
-	// Transport Topology
-	Network Topology
+	Endpoint Topology
+	Address  Topology
```

Comment:
I would love it if you could put a comment about what is in each topology - for instance, when their are nodes, when there are only edges, what the key type is etc.